### ISSUE TITLE
Remote repository produces strange packages for tests

### ISSUE THREAD

**Issue by yarodinD** [spring-cloud/spring-cloud-contract#890]
Spring Boot 2.1.0, Spring Cloud Contract 2.1.0

I tried to use the common repository approach from the documentation: https://cloud.spring.io/spring-cloud-static/spring-cloud-contract/2.1.0.RELEASE/single/spring-cloud-contract.html#_common_repo_with_contracts

The contracts in the example are stored under the servers groupId and artifactId as a path. Just as in the example, I pack the contracts into a jar using assembly plugin.

On the server implementation, I use <contractsMode>REMOTE</contractsMode> in the plugin to fetch the contract jar from nexus. I also define a <baseClassForTests> to inherit the tests from.

The issue is that this leads to very long package names for the tests because it uses the package of the base class, appending the path to the contract.

Example:
Server coordinates: com.example:myserver
Contracts are therefore in com/example/myserver/clientA/expectation.groovy
BaseClassForTests: com.example.server.tests.BaseClass

This results in of the generated tests look like this:
com.example.server.tests.com.example.server.myserver.ClientATest

This works, but the test package is ugly.

My proposal: 
* When baseClassForTests or basePackageForTests are set, use this as a package name and omit the path from the servers groupId:artifactId
* If none of the above is set, use servers maven coordinates as package name (no prefix)
* This is of course for contracts coming from a jar, not from resources/contracts repository.

**Comment by marcingrzejszczak** [comment]
> Remote repository produces strange packages for tests 

It's not strange, it's basing on the folder structure in the packaged jar.

**Comment by yarodinD** [comment]
> > Remote repository produces strange packages for tests
> 
> It's not strange, it's basing on the folder structure in the packaged jar.

True. My bad. "Strange" not in the sense of "I don't know where it comes from" but in the sense of "it looks strange".  :)

**Comment by marcingrzejszczak** [comment]
Hey @yarodinD , can you check if that's also the case in the latest snapshots?

**Comment by yarodinD** [comment]
When I use 2.2.0.BUILD-SNAPSHOT from Aug. 7, now it looks like
com.example.server.clientA.com.example.server.server_artifact_id.clientA.*Test

As far as I can see, the pattern has changed, but patterns are somehow still duplicated.
